gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProc.java

```diff
@@ -55,14 +62,23 @@ public abstract class DagProc<T> {
   @Getter protected final DagManager.DagId dagId;
   @Getter protected final DagNodeId dagNodeId;
   protected static final MetricContext metricContext = Instrumented.getMetricContext(new State(), DagProc.class);
+  protected final List<Class<? extends Exception>> nonRetryableExceptions;
   protected static final EventSubmitter eventSubmitter = new EventSubmitter.Builder(
       metricContext, "org.apache.gobblin.service").build();
 
-  public DagProc(DagTask dagTask) {
+  public DagProc(DagTask dagTask, Config config) {
     this.dagTask = dagTask;
     this.dagId = DagManagerUtils.generateDagId(this.dagTask.getDagAction().getFlowGroup(),
         this.dagTask.getDagAction().getFlowName(), this.dagTask.getDagAction().getFlowExecutionId());
     this.dagNodeId = this.dagTask.getDagAction().getDagNodeId();
+    this.nonRetryableExceptions = ConfigUtils.getStringList(config, ServiceConfigKeys.DAG_PROC_ENGINE_NON_RETRYABLE_EXCEPTIONS_KEY)
```

Comment:
what happens if the config has an empty list?

Reply:
ConfigUtils.getStringList returns Collections.emptyList() if config is not present